rust/otap-dataflow/crates/config/src/pipeline.rs

```diff
@@ -612,11 +622,207 @@ impl Default for PipelineConfigBuilder {
     }
 }
 
+/// Service-level telemetry configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct ServiceConfig {
+    /// The telemetry backend to which to report metrics.
+    #[serde(default)]
+    pub telemetry: TelemetryConfig,
+}
+
+impl Default for ServiceConfig {
+    fn default() -> Self {
+        Self {
+            telemetry: TelemetryConfig::default(),
+        }
+    }
+}
+
+/// Telemetry backend configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct TelemetryConfig {
+    /// The size of the reporting channel.
+    #[serde(default = "default_reporting_channel_size")]
+    pub reporting_channel_size: usize,
+    /// The interval at which metrics are flushed and aggregated by the collector.
+    #[serde(default = "default_flush_interval")]
+    pub flush_interval: std::time::Duration,
+    /// Metrics system configuration.
+    #[serde(default)]
+    pub metrics: MetricsConfig,
+    /// Resource attributes to associate with telemetry data.
+    /// TODO: Support different types of attribute values.
+    #[serde(default)]
+    pub resource: HashMap<String, String>,
+}
+
+impl Default for TelemetryConfig {
+    fn default() -> Self {
+        Self {
+            metrics: MetricsConfig::default(),
+            resource: HashMap::default(),
+            reporting_channel_size: default_reporting_channel_size(),
+            flush_interval: default_flush_interval(),
+        }
+    }
+}
+
+fn default_reporting_channel_size() -> usize {
+    100
+}
+
+fn default_flush_interval() -> std::time::Duration {
+    std::time::Duration::from_secs(1)
+}
+
+/// Opentelemetry Metrics configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct MetricsConfig {
+    /// The list of metrics readers to configure.
+    #[serde(default)]
+    pub readers: Vec<MetricsReaderConfig>,
+}
+
+impl Default for MetricsConfig {
+    fn default() -> Self {
+        Self {
+            readers: Vec::<MetricsReaderConfig>::new(),
+        }
+    }
+}
+
+/// Opentelemetry Metrics Reader configuration.
+#[derive(Debug, Clone, Serialize, JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum MetricsReaderConfig {
+    /// Periodic reader that exports metrics at regular intervals.
+    Periodic(MetricsReaderPeriodicConfig),
+    /// Pull reader that allows on-demand metric collection.
+    Pull {
+        //TODO: Add specific configuration for supported pull readers.
```

Comment:
Note: The reason I say flush_interval is out of place at the higher level is that Pull metrics readers will not have a flush interval so to speak.